src/core/server/saved_objects/migrations/actions/initialize_action.ts

```diff
@@ -44,16 +44,13 @@ export const checkClusterRoutingAllocationEnabledTask =
         flat_settings: true,
       })
       .then((settings) => {
-        const clusterRoutingAllocations: string[] =
-          settings?.transient?.[routingAllocationEnable] ??
-          settings?.persistent?.[routingAllocationEnable] ??
-          [];
+        // if set, only allow the value 'all'
+        const clusterRoutingAllocationEnabledIsAll = [
+          settings?.transient?.[routingAllocationEnable],
+          settings?.persistent?.[routingAllocationEnable],
+        ].every((s: string) => s === undefined || s === 'all');
```

Comment:
Should we really check for the value of each source, instead of just checking the highest priority one, as we're doing here:

https://github.com/elastic/kibana/blob/6627bd8b3ab49415e397495d8e076f9861fcc894/src/core/server/elasticsearch/is_scripting_enabled.ts#L24-L28

Reply:
yeah we could have transient set to `all` and persistent set to `none` which would be a valid configuration.